package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +190,71 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+M.subprocess = {}
+
+M.subprocess.DEVNULL = -1
+M.subprocess.INHERIT = nil
+M.subprocess.PIPE = 1
+
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.subprocess.popen(path, argt, options)
+	local childfds = {}
+	local parentfds = {}
+
+	for iostream, option in pairs(options) do
```

Comment:
I think we should iterate over {'stdin', 'stdout', 'stderr'} explicitly here - `options` may be used for other options later.